src/screens/InitWallet.js

```diff
@@ -265,44 +309,24 @@ class LoadWordsScreen extends React.Component {
     }) });
 
   onChangeText = (text) => {
-    const words = text.trim(/\s+/);
-    let errorMessage = '';
-    let isValid = false;
-    if (words) {
-      try {
-        walletUtils.wordsValid(words);
-        isValid = true;
-      } catch (e) {
-        if (e instanceof hathorErrors.InvalidWords) {
-          errorMessage = e.message;
-          if (e.invalidWords && e.invalidWords.length > 0) {
-            errorMessage = `${errorMessage} List of invalid words: ${e.invalidWords.join(', ')}.`;
-          }
-        } else {
-          throw e;
-        }
-      }
-    }
-
-    const wordsArr = words.split(/\s+/);
-    const nonEmptyWords = wordsArr.filter((value) => value.length !== 0);
-    const nonEmptyWordsLowerCase = nonEmptyWords.map((value) => value.toLowerCase());
+    const { words, isValid, errorMessage } = handleValidWords(text);
     this.setState({
-      words: nonEmptyWordsLowerCase,
+      rawWords: text,
+      words,
       errorMessage,
       isValid,
     });
   }
 
   loadClicked = () => {
     Keyboard.dismiss();
-    const words = this.state.words.join(' ');
-    this.setState({ errorMessage: '' });
-    const result = walletUtils.wordsValid(words);
-    if (result.valid) {
-      this.props.navigation.navigate('ChoosePinScreen', { words });
+    const { rawWords } = this.state;
+    const { isValid, errorMessage } = handleValidWords(rawWords);
+
+    if (isValid) {
+      this.props.navigation.navigate('ChoosePinScreen', { words: rawWords });
```

Comment:
In the old code the parameter was a string with `nonEmptyWordsLowerCase` and now you are using the raw words. Is this still working with multiple spaces between two words or with a mix of upper/lower case?

Reply:
You're right, that was a bug.
It was fixed in https://github.com/HathorNetwork/hathor-wallet-mobile/pull/653/commits/4ad8667946a1107b52f2c4139a8d63b26d3ee0a8 by passing the treated words array joined as a string, like it was before.